tests/unit/app/services/auth.test.js

```diff
@@ -0,0 +1,98 @@
+import axios from 'axios'
+import Auth from 'app/js/services/auth.js'
+import Cookies from 'js-cookie'
+import { Promise } from 'es6-promise'
+
+describe('Auth', () => {
+
+  beforeEach(() => {
+    jasmine.Ajax.install();
+  });
+
+  afterEach(() => {
+    jasmine.Ajax.uninstall();
+  });
+
+  describe('init', () => {
+
+    describe('basic ajax call setup', () => {
+      it('add authorization cookie to request headers', (done)=>{
+        Cookies.set('accessToken', 'secretToken')
+        jasmine.Ajax.stubRequest('/foo').andReturn({status: 200});
+
+        Auth.init();
+
+        axios('/foo').then(() => {
+          let request = jasmine.Ajax.requests.mostRecent();
+          expect(request.requestHeaders.Authorization).toEqual('secretToken');
+          done();
+        });
+      })
+    })
+
+    describe('when accessToken is invalid', () => {
+      beforeEach(() => {
+        Cookies.set('refreshToken', 'secretRefreshToken')
+        jasmine.Ajax.stubRequest('/unauthorized').andReturn({status: 401});
+        jasmine.Ajax.stubRequest('http://test.com/api/v1/api_session/refresh').andReturn({
+          responseText: "{\"access_token\":\"NewAccessToken\"}"
+        });
+
+        Auth.init()
+      })
+
+
+      it('call refresh token API endpoint with refresh token from cookie', (done)=>{
+        axios.get('/unauthorized').then(()=> {
+          let request = jasmine.Ajax.requests.at(1);
+
+          expect(request.url).toEqual('http://test.com/api/v1/api_session/refresh');
+          expect(request.method).toEqual('POST');
+          expect(request.data()).toEqual({ refresh_token: 'secretRefreshToken' });
+
+          done()
+        })
+      })
+
+      it('set new access token to cookie', (done)=>{
+        axios.get('/unauthorized').then(()=> {
+          expect(Cookies.get('accessToken')).toEqual('NewAccessToken');
+          done()
+        })
+      })
+
+      it('recall previous request', (done)=>{
+        axios.get('/unauthorized').then(()=> {
+          let request = jasmine.Ajax.requests.mostRecent();
+          expect(request.url).toEqual('/unauthorized');
+          done()
+        })
+      })
+    })
+
+    fdescribe('when refreshToken is invalid', () => {
```

Comment:
Careful, with `fdescribe` only this set of tests will run, so we can't be sure (or CI can't be sure) that everything is green.